dsgrid/config/project_config.py

```diff
@@ -37,15 +46,127 @@
 from .mapping_tables import MappingTableByNameModel
 from .dimensions import (
     DimensionReferenceModel,
-    DimensionType,
     handle_dimension_union,
     DimensionModel,
+    check_display_name,
+    generate_dimension_query_name,
 )
 
 
 logger = logging.getLogger(__name__)
 
 
+class SubsetDimensionSelectorModel(DSGBaseModel):
+    """Defines a subset dimension selector inclusive of the subset's records and information
+    required to define the selector as a record within the supplemental dimension defined by the
+    subset dimension group.
+    """
+
+    name: str
+    description: str
+    column_values: dict[str, str] = Field(
+        title="column_values",
+        description="Optional columns to populate in the subset dimension group's supplemental "
+        "dimension records table. For example, if each selector in the group defines the end "
+        "uses for one sector (e.g., commercial_end_uses, transportation_end_uses), the "
+        "supplemental dimension records table needs to define the 'fuel_id' and 'unit' fields of "
+        "the EnergyEndUse data model.",
+        default={},
+    )
+    records: list[str] = Field(
+        title="records",
+        description="Table of values populated by reading the parent subset dimension records "
+        "file. Should not be populated by the user.",
+        default=[],
+        dsgrid_internal=True,
+    )
+
+
+class SubsetDimensionGroupModel(DSGBaseModel):
+    """Defines one or more subset dimension selectors for a dimension type."""
+
+    name: str
+    display_name: str
+    dimension_query_name: Optional[str] = Field(
+        title="dimension_query_name",
+        description="Auto-generated query name for SQL queries.",
+    )
+    description: str
+    dimension_type: DimensionType = Field(
+        title="dimension_type",
+        alias="type",
+        description="Type of the dimension",
+        options=DimensionType.format_for_docs(),
+    )
+    filename: Optional[str] = Field(
+        title="filename",
+        alias="file",
+        description="Filename containing dimension records. Only populated for initial "
+        "registration. Each selector's records are stored as JSON objects in the dsgrid registry.",
+    )
+    selectors: list[SubsetDimensionSelectorModel] = Field(
+        title="selectors",
+        description="Dimension selectors",
+    )
+    selector_references: list[DimensionReferenceModel] = Field(
+        title="selectors",
+        description="References to the subset dimensions generated by dsgrid during registration.",
+        default=[],
+    )
+    create_supplemental_dimensions: bool = Field(
```

Comment:
Should this be `create_supplemental_dimension` instead of `create_supplemental_dimensions`? (I think only one supplemental dimension is (optionally) created per SubsetDimensionGroup, not multiple?)

Reply:
true